src/custom_source_buffers/text/html/buffer_manager.ts

```diff
@@ -240,53 +257,136 @@ export default class TextBufferManager {
             cuesInfos.start = end;
             cuesBuffer.splice(i, 0, cuesInfosToInsert);
             return;
+          } else if (areNearlyEqual(end, cuesInfos.end)) {
+            //   ours:            |AAAAAAA|
+            //   the current one:    |BBBB|
+            //   Result:          |AAAAAAA|
+            // Replace
+            cuesBuffer.splice(i, 1, cuesInfosToInsert);
+            return;
+          } else if (end < cuesInfos.end) {
+            //   ours:            |AAAAAAA|
+            //   the current one:     |BBBBB|
+            //   Result:          |AAAAAAABB|
+            cuesInfos.cues = getCuesAfter(cuesInfos.cues, end);
+            cuesInfos.start = end;
+            cuesBuffer.splice(i, 0, cuesInfosToInsert);
+            return;
           }
-          // our cue overlaps the current one:
+
           //   ours:            |AAAAAAA|
-          //   the current one:     |BBBBB|
-          //   Result:          |AAAAAAABB|
-          // Which means:
-          //   1. remove some cues at the start of the current one
-          //   2. update start of current one
-          //   3. add ours before the current one
-          cuesInfos.cues = getCuesAfter(cuesInfos.cues, end);
-          cuesInfos.start = end;
-          cuesBuffer.splice(i, 0, cuesInfosToInsert);
-          return;
+          //   the current one:   |BBB|...
+          //   Result:          |AAAAAAA|...
+          do {
+            cuesBuffer.splice(i, 1);
+            cuesInfos = cuesBuffer[i];
+          } while (cuesInfos !== undefined && end > cuesInfos.end);
+
+          if (
+            cuesInfos === undefined || // There is no cue here
+            areNearlyEqual(end, cuesInfos.end) // this cue has the same end
+          ) {
+            // put in place
+            cuesBuffer[i] = cuesInfosToInsert;
+            return;
+          }
+
+          if (cuesInfos.start >= end) {
```

Comment:
These if() { ... } else { ... } is used three times in the function.
Do you think it is possible to try to factorize this part ? I realize that it is hard to define what this part does, out of the whole function flow. (Otherwise, the logic is good)